src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1475,46 +1475,87 @@ private String normalizeIObjectTemplateName(ObjectType type, String displayName)
     }
 
     private void visitTypeAlias(JSType registryType, TypedVar symbol) {
-      visitTypeAlias(registryType, getUnqualifiedName(symbol));
+      visitTypeAlias(registryType, getUnqualifiedName(symbol), false);
     }
 
-    private void visitTypeAlias(JSType registryType, String unqualifiedName) {
+    private void visitTypeAlias(
+        JSType registryType, String unqualifiedName, boolean emitNeverBrand) {
       emit("type");
       emit(unqualifiedName);
       emit("=");
       visitType(registryType, true, false);
+      // emit a brand to prevent accidental compatibility of values with an enum.
+      if (emitNeverBrand) emit("&{brand: never}");
       emit(";");
       emitBreak();
     }
 
-    private void visitEnumType(String symbolName, EnumType type) {
+    private void visitEnumType(String symbolName, String qualifiedName, EnumType type) {
       // Enums are top level vars, but also declare a corresponding type:
       // <pre>
       // /** @enum {ValueType} */ var MyEnum = {A: ..., B: ...};
       // type MyEnum = EnumValueType;
       // var MyEnum: {A: MyEnum, B: MyEnum, ...};
       // </pre>
-      // TODO(martinprobst): Special case number enums to map to plain TS enums?
-      String unqualifiedName = getUnqualifiedName(symbolName);
+      // We special case "number" enums (and for TS 2.4 should include 'string' enums)
+
       // TS `type` declarations accept only unqualified names.
-      visitTypeAlias(type.getElementsType().getPrimitiveType(), unqualifiedName);
-      emit("var");
-      emit(unqualifiedName);
-      emit(": {");
-      emitBreak();
-      indent();
-      for (String elem : sorted(type.getElements())) {
-        emit(elem);
-        emit(":");
-        // No need to use type.getMembersType(), this must match the type alias we just declared.
+      String unqualifiedName = getUnqualifiedName(symbolName);
+
+      // @enums can be aliased by assignment. Emit a type alias + value alias for the situation.
+      String elementsTypeName = type.getElementsType().getReferenceName();
+      if (!qualifiedName.equals(elementsTypeName)) {
+        emitComment(symbolName + " aliases enum " + elementsTypeName);
```

Comment:
We filter the `// !!` comments before doing test comparisons, to keep test files less noisy. We should probably not do that for comments we emit for the user to see (as opposed to comments we put into the test files). Maybe that's all silly, but it's a change for a different CL :-)